pkg/sql/opt/norm/testdata/rules/inline

```diff
@@ -1343,3 +1343,179 @@ project
  ├── scan a
  └── projections
       └── multi_stmt() [as=multi_stmt:10, immutable, udf]
+
+exec-ddl
+CREATE FUNCTION empty() RETURNS BOOL STABLE LANGUAGE SQL AS ''
+----
+
+# Empty UDFs are not inlined.
+norm expect-not=InlineUDF
+SELECT empty() FROM (VALUES (10), (20)) v(i)
+----
+project
+ ├── columns: empty:2
+ ├── cardinality: [2 - 2]
+ ├── stable
+ ├── fd: ()-->(2)
+ ├── values
+ │    ├── cardinality: [2 - 2]
+ │    ├── ()
+ │    └── ()
+ └── projections
+      └── empty() [as=empty:2, stable, udf]
+
+exec-ddl
+CREATE FUNCTION x_0_1_y_2(x INT, y INT) RETURNS BOOL IMMUTABLE LANGUAGE SQL AS $$
+  SELECT x = 0 OR x = 1 OR y = 2
+$$
+----
+
+# Do not inline a UDF when the parameter corresponding to a subquery argument is
+# referenced multiple times.
+# TODO(mgartner): Fix the formatting of subquery UDF arguments.
+norm expect-not=InlineUDF
+SELECT x_0_1_y_2((SELECT v FROM (VALUES (10), (20)) v(v) WHERE v > a.i), 10) FROM a
```

Comment:
> I haven't looked through carefully yet, but what do you think about just disallowing inlining if any of the arguments is not a variable, constant or placeholder, no matter how many times the parameter is referenced? Then, for cases we care about we can hoist the argument and replace it with a variable to allow inlining.

@DrewKimball thanks for bringing this up. It made me realize this approach is totally flawed. It only checks if an argument expression is a `*memo.Subquery`, so it doesn't catch cases where an argument expression _contains_ a `*memo.Subquery`. 

Adjust this test slightly makes it fail:

```diff
-SELECT x_0_1_y_2((SELECT v FROM (VALUES (10), (20)) v(v) WHERE v > a.i), 10) FROM a
+SELECT x_0_1_y_2(1 + (SELECT v FROM (VALUES (10), (20)) v(v) WHERE v > a.i), 10) FROM a
```

I'm going to close this PR and start again. I'll break out the first commit so we can get that merged now.